### PR TITLE
Fix test parallelism and add some missing test flags

### PR DESCRIPTION
**Pull request by upodroid**
It is looking good so far: https://testgrid.k8s.io/sig-cluster-lifecycle-kubeup-to-kops#Summary

Need to run the conformance and serial with parallelism set to one.

I'm also experimenting with running conformance tests in parallel and `--ginkgo.flake-attempts` set to 3 to rerun serial/disruptive tests that will blip when run in parallel

### CODE REVIEW

**Comment by aojea** [comment]
No flake attempt, we have zero tolerance to flakes, test must not retry

File: config/jobs/kubernetes/kops/build_jobs.py [hunk 1]
```diff
@@ -889,10 +895,30 @@ def generate_misc():
...
+                   kops_version="https://storage.googleapis.com/kops-ci/bin/latest-ci.txt",
+                   kops_channel="alpha",
+                   build_cluster="k8s-infra-prow-build",
+                   extra_flags=[
+                       "--image=cos-cloud/cos-105-17412-156-49",
+                   ],
+                   focus_regex=r'\[Conformance\]',
+                   skip_regex=r'\[FOOBAR\]', # leaving it empty will allow kops to add extra skips
+                   test_timeout_minutes=100,
+                   test_args="-num-nodes=3 --master-os-distro=gci --node-os-distro=gci --ginkgo.flake-attempts=3", # pylint: disable=line-too-long
```
Comment (aojea) [inline config/jobs/kubernetes/kops/build_jobs.py:918]: That is not valid for our CI, we removed flake attempt some time ago because it hides bugs

Comment (aojea) [reply config/jobs/kubernetes/kops/build_jobs.py:918]: please remove flake-attempts, there has to be zero tolerance to flakes, I sent an email and updated our docs with that ... the rest lgtm

Comment (upodroid) [reply config/jobs/kubernetes/kops/build_jobs.py:918]: It is an entirely new job, I have a proper conformance job running with no flake-attempts allowed and parallel of 1.

Comment (aojea) [reply config/jobs/kubernetes/kops/build_jobs.py:918]: yeah, but why should we allow retries on flakes?

Comment (upodroid) [reply config/jobs/kubernetes/kops/build_jobs.py:918]: Looking at the board, it seems that a small set of conformance tests struggle with high parallelism. 

Conformance test in serial takes 2h40m
Conformance test in parallel(30) takes around 40m, but at the cost of a few flakes from the tests not being robust enough.


Is this worth investigating or not?

Comment (upodroid) [reply config/jobs/kubernetes/kops/build_jobs.py:918]: For example, presubmits can run "quick conformance" tests and the existing "legit" conformance test

**Pull request merged**